### PR TITLE
JHy/build-improvements

### PR DESCRIPTION
**Pull request by janhybs**
1) Build directory is now within the main dir:
  - was: `../build-$GIT_BRANCH`
  - is: `build-$GIT_BRANCH`

2) Fix some issue with newer python libs
3) Add option to specify output path of a profiler via `--profiler-path foo.json`

### CODE REVIEW

File: src/system/sys_profiler.cc [hunk 1]
```diff
@@ -714,6 +714,23 @@ void Profiler::output(MPI_Comm comm) {
     }
 }
 
+void Profiler::output(MPI_Comm comm, string profiler_path) {
```
Comment (jbrezmorf) [inline src/system/sys_profiler.cc]: Add the `profiler_path` parameter as optional to the existing methods.

File: src/python/scripts/yamlc/yaml_config.py [hunk 2]
```diff
@@ -175,7 +175,10 @@ def get_one(self, yaml_case_file):
 
     def _read_yaml(self):
         with open(self.yaml_config_file, 'r') as fp:
-            result = yaml.load(fp)
+            try:
+                result = yaml.safe_load(fp)
```
[the remark this answers is not in the document]
Comment (janhybs) [reply src/python/scripts/yamlc/yaml_config.py]: Simple `load` will be deprecated in the future due to security concerns.

File: src/system/sys_profiler.cc (hunk 1 above)
Comment (janhybs) [reply src/system/sys_profiler.cc]: done

File: src/python/scripts/yamlc/yaml_config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janhybs) [reply src/python/scripts/yamlc/yaml_config.py]: Simple `load` will be deprecated in the future due to security concerns.

**Comment by janhybs** [comment]
Since `CMakeLists.txt` is related to the build process, this branch now includes default `config.default.cmake` which will not require any `config.cmake` when running inside docker `flow-libs-dev-rel` or `flow-libs-dev-rel`

File: src/python/scripts/yamlc/yaml_config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbrezmorf) [reply src/python/scripts/yamlc/yaml_config.py]: I know, so why not just use the safe variant.

File: src/system/sys_profiler.cc [hunk 3]
```diff
@@ -703,14 +701,20 @@ void Profiler::output(MPI_Comm comm, ostream &os) {
...
-    int mpi_rank;
-    chkerr( MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank) );
-    if (mpi_rank == 0) {
-        output(comm, *get_default_output_stream());
-    } else {
-        ostringstream os;
-        output(comm, os );
+void Profiler::output(MPI_Comm comm, string profiler_path /* = "" */) {
+    if(profiler_path == "") {
+        output(comm);
```
Comment (jbrezmorf) [inline src/system/sys_profiler.cc]: `if (mpi_rank == 0)` should come first.
then use `*get_default_output_stream()` in the case of empty path.

**Comment by jbrezmorf** [comment]
> Since `CMakeLists.txt` is related to the build process, this branch now includes default `config.default.cmake` which will not require any `config.cmake` when running inside docker `flow-libs-dev-rel` or `flow-libs-dev-dbg`

Any problem with that?

File: src/system/sys_profiler.cc (hunk 3 above)
Comment (janhybs) [reply src/system/sys_profiler.cc]: Done

**Pull request merged**